Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
Running any of those with `--debug` should help you get a backtrace. Can you paste that here so we can figure out the best place to put this call in the stack? Probably doesn't need to be in `brew.rb` in that case.